query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapQuerySpec.groovy

```diff
@@ -0,0 +1,242 @@
+    }
+
+    def 'Fail to connect to LDAP'() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().port(666).getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().getValue(),
+                (LdapQuery.QUERY_BASE_FIELD_NAME)       : exampleQueryBase().getValue(),
+                (LdapQueryField.DEFAULT_FIELD_NAME)     : exampleLdapQuery().getValue()]
+        ldapQueryFunction.setValue(args)
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages().get(0).getCode() == DefaultMessages.CANNOT_CONNECT
+        report.messages().get(0).getPath() == [LdapQuery.FIELD_NAME, BaseFunctionField.ARGUMENT, LdapConnectionField.DEFAULT_FIELD_NAME]
+
+    }
+
+    def 'Fail to bind user over bind method "Simple" with bad password'() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().password("badPassword").getValue(),
+                (LdapQuery.QUERY_BASE_FIELD_NAME)       : exampleQueryBase().getValue(),
+                (LdapQueryField.DEFAULT_FIELD_NAME)     : exampleLdapQuery().getValue()]
+
+        ldapQueryFunction.setValue(args)
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages().get(0).getCode() == LdapMessages.CANNOT_BIND
+        report.messages().get(0).getPath() == [LdapQuery.FIELD_NAME, BaseFunctionField.ARGUMENT, LdapBindUserInfo.DEFAULT_FIELD_NAME]
+    }
+
+    def 'Successfully query with entries found (No encryption)'() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): noEncryptionLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().password("secret").getValue(),
+                (LdapQuery.QUERY_BASE_FIELD_NAME)       : exampleQueryBase().getValue(),
+                (LdapQueryField.DEFAULT_FIELD_NAME)     : ldapQuery("(objectClass=person)").getValue()]
+
+        ldapQueryFunction.setValue(args)
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        def entries = report.result().getList()
+        entries.size() == 2
+        entries*.getEntry('name')*.get()*.value() as Set ==
+                ['uid=tstark,ou=users,dc=example,dc=com', 'uid=bbanner,ou=users,dc=example,dc=com'] as Set
+        entries*.getEntry('employeeType')*.get()*.value() as Set == ['ironman', 'hulk'] as Set
+    }
+
+    def 'Successfully query with entries found (LDAPS) '() {
+        setup:
+        args = [(LdapConnectionField.DEFAULT_FIELD_NAME): ldapsLdapConnectionInfo().getValue(),
+                (LdapBindUserInfo.DEFAULT_FIELD_NAME)   : simpleBindInfo().password("secret").getValue(),
```

Comment:
It's not retrieving the password, it's setting it then turning bindInfo into a map.